apps/spruce/src/pages/adminSettings/tabs/GeneralTab/OtherTab/schemaFields.ts

```diff
@@ -0,0 +1,546 @@
+    },
+    uiSchema: {
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      "ui:data-cy": "single-task-host",
+      "ui:objectFieldCss": objectGridCss,
+      projectTasksPairs: {
+        "ui:addButtonText": "Add project tasks pair",
+        "ui:data-cy": "project-tasks-pairs-list",
+        "ui:orderable": false,
+        "ui:fullWidth": true,
+        "ui:fieldCss": fullWidthCss,
+        "ui:arrayItemCSS": arrayItemCSS,
+        items: {
+          allowedTasks: {
+            "ui:widget": widgets.ChipInputWidget,
+          },
+          allowedBVs: {
+            "ui:widget": widgets.ChipInputWidget,
+          },
+        },
+      },
+    },
+  };
+};
+
+export const bucketConfig = {
+  schema: {
+    logBucket: {
+      type: "object" as const,
+      title: "",
+      properties: {
+        defaultLogBucket: {
+          type: "string" as const,
+          title: "Default Log Bucket",
+        },
+        logBucketLongRetentionName: {
+          type: "string" as const,
+          title: "Long Retention Log Bucket",
+        },
+        testResultsBucketName: {
+          type: "string" as const,
+          title: "Test Results Bucket Name",
+        },
+        testResultsBucketTestResultsPrefix: {
+          type: "string" as const,
+          title: "Test Results Bucket Prefix",
+        },
+        testResultsBucketRoleARN: {
+          type: "string" as const,
+          title: "Test Results Bucket Role ARN",
+        },
+        credentialsKey: {
+          type: "string" as const,
+          title: "S3 Key",
+        },
+        credentialsSecret: {
+          type: "string" as const,
+          title: "S3 Secret",
+        },
+        longRetentionProjects: {
+          type: "array" as const,
+          title: "Projects Requiring Long Retention",
+          items: {
+            type: "string" as const,
+          },
+        },
+      },
+    },
+  },
+  uiSchema: {
+    logBucket: {
+      "ui:ObjectFieldTemplate": CardFieldTemplate,
+      "ui:data-cy": "bucket-config",
+      "ui:objectFieldCss": objectGridCss,
+      longRetentionProjects: {
+        "ui:widget": widgets.ChipInputWidget,
+        "ui:fieldCss": fullWidthCss,
+      },
+    },
+  },
+};
+
+export const sshPairs = {
+  schema: {
+    taskHostKey: {
+      type: "object" as const,
+      title: "Task Host Key",
+      properties: {
+        name: {
+          type: "string" as const,
+          title: "Name",
+        },
+        secretARN: {
+          type: "string" as const,
+          title: "Secret ARN",
+        },
+      },
+    },
+    spawnHostKey: {
+      type: "object" as const,
+      title: "Spawn Host Key",
+      properties: {
+        name: {
+          type: "string" as const,
+          title: "Name",
+        },
+        secretARN: {
+          type: "string" as const,
+          title: "Secret ARN",
+        },
+      },
+    },
+    kanopySSHKeyPath: {
+      type: "string" as const,
+      title: "Legacy SSH Key",
+    },
```

Comment:
Maybe would suggest moving this above the other fields so it doesn't look like it's a part of `Task Host Key` or `Spawn Host Key`